antarest/study/service.py

```diff
@@ -284,7 +319,7 @@ def _upgrade_study(self) -> None:
                     file_study = self.storage_service.get_storage(study_to_upgrade).get_raw(study_to_upgrade)
                     file_study.tree.normalize()
 
-    def run_task(self, notifier: TaskUpdateNotifier) -> TaskResult:
+    def run_task(self, notifier: TaskUpdateNotifier, listener: t.Optional[ICommandListener]) -> TaskResult:
```

Comment:
We should not need to pass a command listener to all types of tasks, because they don't necessarily need to execute commands.

We need to separate the 2 levels of listeners to be consistent with the task -> commands stack:
- The task listener
- The command listener

Tasks can apply commands, they are higher in the stack.
If we need to have notifications travel up the stack, a command listener can forward to a task listener.

So in practice here, I think we should still have only one argument, either the existing notifier or a TaskListener whhich would have a new "notify_progress" method.

Then, down the stack, when we generate a timeseries, we will create an implementation of ICommandListener which will forward the progress up the stack to the task listener/notifier.